--- ./sys-utils/losetup.c-orig	2019-10-08 13:11:53.759155901 +0200
+++ ./sys-utils/losetup.c	2020-08-18 16:47:57.111449949 +0200
@@ -594,7 +594,7 @@ int main(int argc, char **argv)
 	int res = 0, showdev = 0, lo_flags = 0;
 	char *outarg = NULL;
 	int list = 0;
-	unsigned long use_dio = 0, set_dio = 0, set_blocksize = 0;
+	unsigned long use_dio = 1, set_dio = 1, set_blocksize = 0;
 
 	enum {
 		OPT_SIZELIMIT = CHAR_MAX + 1,
@@ -723,7 +723,7 @@ int main(int argc, char **argv)
 		case OPT_DIO:
 			use_dio = set_dio = 1;
 			if (optarg)
-				use_dio = parse_switch(optarg, _("argument error"), "on", "off", NULL);
+				use_dio = set_dio = parse_switch(optarg, _("argument error"), "on", "off", NULL);
 			break;
 		case 'v':
 			break;
