From 970d9af9015a387bb81841faf05dcc1a171eb97a Mon Sep 17 00:00:00 2001
From: Philip Prindeville <philipp@redfish-solutions.com>
Date: Sun, 1 Jan 2023 15:25:04 -0700
Subject: [PATCH v3 1/1] x86: Support APU5 in PCEngines platform driver

--- a/drivers/clocksource/timer-cs5535.c
+++ b/drivers/clocksource/timer-cs5535.c
@@ -127,7 +127,9 @@ static irqreturn_t mfgpt_tick(int irq, v
 		cs5535_mfgpt_write(cs5535_event_clock, MFGPT_REG_SETUP,
 				MFGPT_SETUP_CNTEN | MFGPT_SETUP_CMP2);
 
-	cs5535_clockevent.event_handler(&cs5535_clockevent);
+	if (cs5535_clockevent.event_handler)
+		cs5535_clockevent.event_handler(&cs5535_clockevent);
+
 	return IRQ_HANDLED;
 }
 
